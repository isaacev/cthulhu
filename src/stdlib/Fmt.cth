#[inline]
fn int(i: Int) -> Str {
  ::Kernel::Builtins::cast_int_to_string(i)
}

#[inline]
fn float(f: Float) -> Str {
  ::Kernel::Builtins::cast_float_to_string(f)
}
