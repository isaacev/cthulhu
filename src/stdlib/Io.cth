intrinsic {
  fn php_print(Str) -> ();
  fn read_argv() -> [Str];
}

#[inline]
pub fn print(str: Str) -> () {
  php_print(str)
}

#[inline]
pub fn println(str: Str) -> () {
  php_print(str ++ "\n")
}

pub fn argv() -> [Str] {
  read_argv()
}
