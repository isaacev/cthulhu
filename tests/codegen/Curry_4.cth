use ::Io;
use ::Fmt;

fn inner(d: Int, e: Int, f: Int) -> Int {
  Io::println("enter inner");
  (d * e) + f
}

fn foo(a: Int, b: Int) -> Int -> Int {
  Io::println("enter foo");
  inner(a, b)
}

#[entry]
fn main() -> () {
  let x = foo(1, 2, 3);
  Io::println(Fmt::int(x));
}
