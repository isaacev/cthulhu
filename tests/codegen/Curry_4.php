<?php

namespace Curry_4 {
  function inner($d, $e, $f) {
    print("enter inner" . "\n");
    return $d * $e + $f;
  }
  function foo($a, $b) {
    print("enter foo" . "\n");
    return fn ($d) => \Curry_4\inner($a, $b, $d);
  }
  function main() {
    $x = \Curry_4\foo(1, 2)(3);
    print((string)$x . "\n");
  }
}

namespace {
  \Curry_4\main(null);
}
