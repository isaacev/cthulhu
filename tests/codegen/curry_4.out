enter foo
enter inner
5
